resolver/resolver.cc

```diff
@@ -129,13 +129,19 @@ class ResolveConstantsWalk {
 
         Nesting(shared_ptr<Nesting> parent, core::SymbolRef scope) : parent(std::move(parent)), scope(scope) {}
     };
+
     shared_ptr<Nesting> nesting_;
+    shared_ptr<UnorderedMap<core::SymbolRef, core::LocOffsets>> localFirstDefOfManySymTable;
```

Comment:
I was initially confused by the "local" here; I guess the "local" is trying to connote "local to a given file"?  I think that's a little hard to parse out, given that there's no reference to files in this data structure.

The name of this symbol is a mouthful.  Maybe just `firstDefinitionLoc` or something, since I think that's what you're trying to capture...and that differs from the loc stored in the symbol table because the canonical loc in the symbol table is the latest one (?).  A comment might help describe why we're tracking this separately from symbol table locs.

I think we could do this without `shared_ptr`, but managing the pointers gets a little more complicated at that point, and I don't think that it's worth it.